lib/convection/control/cloud.rb

```diff
@@ -16,17 +16,23 @@ def stacks
         @cloudfile.stacks
       end
 
-      def deck
-        @cloudfile.deck
+      def stack_groups
+        @cloudfile.stack_groups
       end
 
-      def converge(to_stack, &block)
-        unless to_stack.nil? || stacks.include?(to_stack)
-          block.call(Model::Event.new(:error, "Stack #{ to_stack } is not defined", :error)) if block
-          return
-        end
+      def filter_deck(options = {}, &block)
+        included_stacks = included_stacks(options)
+        return @cloudfile.deck if included_stacks.empty?
```

Comment:
This reads weird:  "No included stacks, OK, I'll return all stacks".  Also there a are a few error conditions that aren't caught:
* user specifies a non-existent stack group -> converges all stacks
* user specifies stack group and stack list -> converges stack group

Would something like this work:

```ruby
# throw an error if the user specifies both a stack group and list of stacks
if options[:stack_group] && options[:stack_list]
  block.call(Model::Event.new(:error, "Cannot specify --stack-group and --stack-list at the same time", :error)) if block
  return []
end

# throw an error if the user specifies a nonexistent stack groups
if (options[:stack_group] && !stack_groups.key?(options[:stack_group]))
  block.call(Model::Event.new(:error, "Unknown stack group: #{options[:stack_group]}", :error)) if block
  return []
end

# throw an error if the user specifies nonexistent stacks
if Array(options[:stack_list]).any? { |name| !@cloudfile.stacks.key?(name) }
  bad_stack_names = options[:stack_list].reject { |name| @cloudfile.stacks.key?(name) }
  block.call(Model::Event.new(:error, "Stack#{'s' if bad_stack_names.length > 1} #{bad_stack_names.join(', ')} #{bad_stack_names.length > 1 ? 'do' : 'does'} not exist", :error)) if block
  return []
end

filter = Array(stack_groups[options[:stack_group]] || options[:stack_list])

# if no filter is specified, return the entire deck
return @cloudfile.deck if filter.empty?

@cloudfile.stacks.select { |name, stack| filter.include?(name) }
```